src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.String.cs

```diff
@@ -9,6 +9,353 @@ namespace System.Text.Json
 {
     public sealed partial class Utf8JsonWriter
     {
+        /// <summary>
+        /// Writes the pre-encoded property name (as a JSON string) as the first part of a name/value pair of a JSON object.
+        /// <param name="propertyName">The JSON encoded property name of the JSON object to be transcoded and written as UTF-8.</param>
+        /// </summary>
+        /// <remarks>
+        /// The property name should already be escaped when the instance of <see cref="JsonEncodedText"/> was created.
+        /// </remarks>
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if this would result in an invalid JSON to be written (while validation is enabled).
+        /// </exception>
+        public void WritePropertyName(JsonEncodedText propertyName)
+            => WritePropertyNameHelper(propertyName.EncodedUtf8Bytes);
+
+        private void WritePropertyNameHelper(ReadOnlySpan<byte> utf8PropertyName)
+        {
+            Debug.Assert(utf8PropertyName.Length <= JsonConstants.MaxTokenSize);
```

Comment:
Is an assert right for this? Does `JsonEncodedText` provide this invariant?

Reply:
Yes it does.
https://github.com/dotnet/corefx/blob/f528a05ef02def705ac9e69b919388e5be0f00ec/src/System.Text.Json/src/System/Text/Json/JsonEncodedText.cs#L97